---
title: "sequence_processing"
author: "Scott Klasek"
date: "1/27/2020"
output: github_document
---

### dada2 sequence processing
Notes from previous document
forward quality scores: the extraction blanks and samples with low reads (like <1000) are really bad, other samples drop off at ~230
reverse quality scores: same pattern, but drop off ~220
my f primer is 19 bp, reverse is 20: forward, GTGCCAGCMGCCGCGGTAA; reverse, TAATCTWTGGGVHCATCAGG

```{r cache=TRUE}
# libraries and environment
library(dada2)
library(tidyr)
library(plyr)
library(dplyr)
library(ggplot2)
library(phyloseq)
library(decontam)
library(DECIPHER)
library(phangorn)
sessioninfo <- sessionInfo()
# dada2
fqpath <- "/Users/scottklasek/Desktop/flux_resubmission/fastq"
original_file_names <- list.files(fqpath)
original_file_names
new_file_names <- list.files(fqpath)
new_file_names
fnFs.2 <- sort(list.files(fqpath, pattern="_R1.fastq", full.names = TRUE))
fnRs.2 <- sort(list.files(fqpath, pattern="_R2.fastq", full.names = TRUE))
sample.names.2 <- sapply(strsplit(basename(fnFs.2), "_"), `[`, 1)
sample.names.2
plotQualityProfile(fnFs.2[30:35])
plotQualityProfile(fnRs.2[30:35])
# some of our samples are sequenced twice. which fastq files are better quality?
dup_samples <- c("GC1048-305.0A","GC1048-305.0B","GC1070-063.0A","GC1070-063.0B","GC1070-068.0A","GC1070-068.0B")
dup_vectors <- match(dup_samples,sample.names.2)
plotQualityProfile(fnFs.2[dup_vectors])
plotQualityProfile(fnRs.2[dup_vectors])
filtFs.2 <- file.path(fqpath, "filtered", paste0(sample.names.2, "_F_filt.fastq"))
filtRs.2 <- file.path(fqpath, "filtered", paste0(sample.names.2, "_R_filt.fastq"))
names(filtFs.2) <- sample.names.2
names(filtRs.2) <- sample.names.2
out.2 <- filterAndTrim(fnFs.2, filtFs.2, fnRs.2, filtRs.2, truncLen=c(230,230), trimLeft=c(19,20),
                       maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
                       compress=TRUE, multithread=TRUE)
head(out.2)
errF.2 <- learnErrors(filtFs.2, multithread=TRUE)
errR.2 <- learnErrors(filtRs.2, multithread=TRUE)
plotErrors(errF.2, nominalQ=TRUE)
plotErrors(errR.2, nominalQ=TRUE)
dadaFs.2 <- dada(filtFs.2, err=errF.2, multithread=TRUE)
dadaRs.2 <- dada(filtRs.2, err=errR.2, multithread=TRUE)
dadaFs.2[[75]]
dadaRs.2[[75]]
mergers.2 <- mergePairs(dadaFs.2, filtFs.2, dadaRs.2, filtRs.2, verbose=TRUE)
# quick trim
seqtab.2 <- makeSequenceTable(mergers.2)
dim(seqtab.2)
table(nchar(getSequences(seqtab.2)))
# The V4 amplicon should be ~291 bp, so trimming each F or R read to 230 bp and removing primers (19-20 bp) should mean that the amplicons # are closer to 252 bp, right? But I remember they were ~252 bp when I hadn't removed the primers when processing these in mothur. Most of # the sequences are 214 bp, 214 + 39 (length of trimmed primers) is 253. So I think ~214 bp is the correct length. 
seqtab.trim.2 <- seqtab.2[,nchar(colnames(seqtab.2)) %in% 212:217]
table(nchar(getSequences(seqtab.trim.2)))
seqtab.nochim.2 <- removeBimeraDenovo(seqtab.trim.2, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim.2)
dim(seqtab.trim.2)
sum(seqtab.nochim.2)/sum(seqtab.trim.2)
getN <- function(x) sum(getUniques(x))
track.2 <- cbind(out.2, sapply(dadaFs.2, getN), sapply(dadaRs.2, getN), sapply(mergers.2, getN), rowSums(seqtab.nochim.2))
colnames(track.2) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track.2) <- sample.names.2
head(track.2)
tdf.2 <- as.data.frame(track.2)
tdf.2$filtered_out <- tdf.2$input-tdf.2$filtered
tdf.2$noise_removed <- tdf.2$filtered-with(tdf.2, pmin(denoisedF, denoisedR))
tdf.2$unmerged <- (tdf.2$filtered-tdf.2$noise_removed)-tdf.2$merged
tdf.2$chimeras <- tdf.2$merged-tdf.2$nonchim
tdf.2 <- data.frame(sample = row.names(tdf.2), tdf.2)
# select only the columns we want to plot:
tdfs.2 <- tdf.2[,c(1,7,8,9,10,11)]
tdfs.2  
tdfl.2 <- gather(tdfs.2, step, reads, nonchim:chimeras, factor_key=FALSE)
tdfl.2
# reorder the steps to plot them in an order that makes sense:
tdfl.2$step <- factor(tdfl.2$step, levels = c("filtered_out","noise_removed","unmerged","chimeras", "nonchim"))
track.reads.2 <- ggplot(tdfl.2,aes(sample,reads,fill=step))
track.reads.plot.2 <- track.reads.2+
  geom_bar(stat="identity")+
  scale_y_continuous(breaks = c(50000,100000,150000,200000,250000))+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 60, hjust = 1))
track.reads.plot.2
taxa.2 <- assignTaxonomy(seqtab.nochim.2, "~/Desktop/flux_resubmission/silva_nr_v132_train_set.fa", multithread=TRUE)
taxa.print.2 <- taxa.2
rownames(taxa.print.2) <- NULL
head(taxa.print.2) 
# note I didn't do any species assignment because 1) I didn't for my mothur run, 2) only 13% of genera were identified anyway (see below), and 
# 3) I got a weird error message during the addSpecies command: "Error: vector memory exhausted (limit reached?)"
message("% phyla identified is ", (1-(sum(is.na(taxa.print.2[,2]))/length(taxa.print.2[,2])))*100)
message("% classes identified is ", (1-(sum(is.na(taxa.print.2[,3]))/length(taxa.print.2[,3])))*100)
message("% orders identified is ", (1-(sum(is.na(taxa.print.2[,4]))/length(taxa.print.2[,4])))*100)
message("% families identified is ", (1-(sum(is.na(taxa.print.2[,5]))/length(taxa.print.2[,5])))*100)
message("% genera identified is ", (1-(sum(is.na(taxa.print.2[,6]))/length(taxa.print.2[,6])))*100)
# 88% of phyla, 76% of classes, 52% of orders, 31% of families, 13% of genera identified in the sediments. compare to 84.5% of families, 60.9% of genera, and 3.9% species identified in rhizosphere communities
```

### Make phylogenetic tree
```{r cache=TRUE}
library(dada2)
library(tidyr)
library(plyr)
library(dplyr)
library(ggplot2)
library(phyloseq)
library(decontam)
library(DECIPHER)
library(phangorn)
seqs <- getSequences(seqtab.nochim.2)
names(seqs) <- seqs
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA) # this took ~50 minutes
phang.align <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phang.align) # started 3:05 pm, finished around an hour later? 
treeNJ <- NJ(dm) # Note, tip order != sequence order ## this takes at least an hour... started 6:01 pm
fit <- pml(treeNJ, data=phang.align)
fitGTR <- update(fit, k=4, inv=0.2)
```

### Add in metadata and create phyloseq object
```{r}
metadata.2 <- read.csv(file="~/Desktop/svalflux/metadata.csv")
rownames(metadata.2) <- sample.names.2
metadata.2
library(phyloseq)
ps.f <- phyloseq(tax_table(taxa.2),
                 sample_data(metadata.2),
                 phy_tree(fitGTR$tree),
                 otu_table(seqtab.nochim.2, taxa_are_rows = FALSE))
dna <- Biostrings::DNAStringSet(taxa_names(ps.f))
names(dna) <- taxa_names(ps.f)
ps.f <- merge_phyloseq(ps.f, dna)
taxa_names(ps.f) <- paste0("ASV", seq(ntaxa(ps.f)))
ps.f # the first phyloseq object
saveRDS(ps.f, "ps.f") # saved phyloseq object, pushed to Github (need to update this, as metadata now has ddPCR data)
```

### Pruning and decontamination
```{r}
library(phyloseq)
ASV_classifications <- c("Bacteria","Archaea","Eukaryota", "Unclassified", "Chloroplast","Mitochondria")
num_ASVs <- c(sum(tax_table(ps.f)[,1]=="Bacteria", na.rm = TRUE),
               sum(tax_table(ps.f)[,1]=="Archaea", na.rm=TRUE),
               sum(tax_table(ps.f)[,1]=="Eukaryota", na.rm=TRUE),
               sum(is.na(tax_table(ps.f)[,1])),
               sum(tax_table(ps.f)[,4]=="Chloroplast", na.rm=TRUE), 
               sum(tax_table(ps.f)[,5]=="Mitochondria", na.rm=TRUE))
num_ASVcounts <- c(sum(otu_table(ps.f)[,which(tax_table(ps.f)[,1]=="Bacteria")], na.rm = TRUE),
              sum(otu_table(ps.f)[,which(tax_table(ps.f)[,1]=="Archaea")], na.rm = TRUE),
              sum(otu_table(ps.f)[,which(tax_table(ps.f)[,1]=="Eukaryota")], na.rm = TRUE), 
              sum(is.na(tax_table(ps.f)[,1])),
              sum(otu_table(ps.f)[,which(tax_table(ps.f)[,4]=="Chloroplast")], na.rm = TRUE), 
              sum(otu_table(ps.f)[,which(tax_table(ps.f)[,5]=="Mitochondria")], na.rm = TRUE))
asv.table <- cbind.data.frame(ASV_classifications, num_ASVs, num_ASVcounts)
asv.table
ps.fr <- subset_taxa(ps.f, (Kingdom!="Eukaryota")) # unknown and eukaryote ASVs removed
ps.fr <- subset_taxa(ps.fr, (Order!="Chloroplast") | is.na(Order)) # chloroplasts removed
ps.fr <- subset_taxa(ps.fr, (Family!="Mitochondria") | is.na(Family)) # mitochondria removed
ntaxa(ps.f)-ntaxa(ps.fr) # number of taxa removed 
ps.fr

# run decontam using the combined approach, which combines the scores from prevalence and frequency using Fisher's method 
library(decontam) 
sample_data(ps.fr)$is.neg <- sample_data(ps.fr)$Sample_or_Control == "Control Sample"
contamdf.flux.comb <- isContaminant(ps.fr, method="combined", neg="is.neg", conc="quant_reading")
table(contamdf.flux.comb$contaminant) # prints the number of contaminants
flux.comb.contam <- (which(contamdf.flux.comb$contaminant))
flux.comb.contam.mx <- subset(tax_table(ps.fr)[flux.comb.contam,])

# plot library sizes of true samples and blanks
library(ggplot2)
df.flux <- as.data.frame(sample_data(ps.fr)) 
df.flux$LibrarySize <- sample_sums(ps.fr)
df.flux <- df.flux[order(df.flux$LibrarySize),]
df.flux$Index <- seq(nrow(df.flux))
flux.sample.depth.samples.and.controls <- ggplot(data=df.flux, aes(x=Index, y=LibrarySize, color=Sample_or_Control)) + geom_point() + ggtitle("Before decontamination")
flux.sample.depth.samples.and.controls 

# remove the contaminants
contam <- rownames(flux.comb.contam.mx) # the contaminant names
allnames <- rownames(tax_table(ps.fr)) # the taxa names
flux.uncontam <- allnames[!allnames %in% contam] # gives us the noncontaminant names
ps.frd <- prune_taxa(flux.uncontam, ps.fr) # creates new phyloseq object without contaminants
ps.frd 

# remembered there were Micrococcus in some groups... 
which(flux.comb.contam.mx[,6]=="Micrococcus") # not detected by decontam
sum(otu_table(ps.fr)[,which(tax_table(ps.fr)[,6]=="Micrococcus")])/sum(otu_table(ps.fr))*100 # 0.67% of sequences
ps.blanks <- subset_samples(ps.fr, Sample_or_Control=="Control Sample") # phyloseq object with only the blanks 
blanks.top100 <- names(sort(taxa_sums(ps.blanks), decreasing=TRUE))[1:100]
blanks.ps.ra <- transform_sample_counts(ps.blanks, function(OTU) OTU/sum(OTU))
blanks.ps.top100 <- prune_taxa(blanks.top100, blanks.ps.ra)
blanks_barplot <- plot_bar(blanks.ps.top100, fill="Class")
blanks_barplot # yes indeed these blanks are different
df.flux["EBkati-000.0A","LibrarySize"] # over 9k reads
EBkati.ra <- get_taxa(blanks.ps.ra, "EBkati-000.0A") # extract relative abundances from one of our blanks
EBkati.ra <- subset(EBkati.ra, EBkati.ra>0) # omit ASVs not detected in the blank
EBkati.ra <- sort(EBkati.ra, decreasing = TRUE) # sort the list of ASVs by descending order
tax_table(ps.blanks)[names(EBkati.ra)[EBkati.ra>0],] # lists taxonomies of ASVs in the sample by descending relative abundance. Here we see the two most abundant ASVs belong to micrococcus. What is the abundance of Micrococcus across all samples?
ps.fr.ra <- transform_sample_counts(ps.fr, function(OTU) OTU/sum(OTU)) # first trasform ps.fr to relative abundances
micrococcus <- subset_taxa(ps.fr.ra, Genus=="Micrococcus")
plot_bar(micrococcus) # only the 1048 201, 260, 261 depths also have Micrococcus in significant amounts, ~20%. They all have >25k reads:

# plot library sizes of true samples and blanks after decontamination
df.dc.flux <- as.data.frame(sample_data(ps.frd)) 
df.dc.flux$LibrarySize <- sample_sums(ps.frd)
df.dc.flux <- df.dc.flux[order(df.dc.flux$LibrarySize),]
df.dc.flux$Index <- seq(nrow(df.dc.flux))
flux.dc.sample.depth.samples.and.controls <- ggplot(data=df.dc.flux, aes(x=Index, y=LibrarySize, color=Sample_or_Control)) + geom_point() + ggtitle("After decontamination")
flux.dc.sample.depth.samples.and.controls # now the other negative control is lower

df.dc.flux["GC1048-201.0A","LibrarySize"]
df.dc.flux["GC1048-260.0A","LibrarySize"]
df.dc.flux["GC1048-261.0A","LibrarySize"]
ps.frd <- subset_taxa(ps.frd, (Genus!="Micrococcus") | is.na(Genus)) # removed 4 Micrococcus ASVs from ps.frd
ps.frd

# prune samples with low-abundance reads
df.dc.flux[10:15,16] # inspect library sizes near the range where we want to prune
ps.frdp <- prune_samples(sample_sums(ps.frd)>=8931, ps.frd) # 12 samples with less than 8931 reads removed (max library size removed was 3816)
ps.frdp # PhyloSeq object Flux, Removed unwanted taxa, Decontaminated, Pruned
saveRDS(ps.frdp, "ps.frdp") # saved phyloseq object, pushed to Github (need to update with metadata)
```

### Import  ps.frdp and add in more metadata
SMT depth, peakAOM depth, distance above/below peakAOM depth, and classification of samples as above or below SMT
```{r}
library(phyloseq)
ps.frdp <- readRDS(file="ps.frdp")

sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1048"),8] <- "ss" # overwrites "inc" to "ss" for core GC1048
sample_data(ps.frdp)$smt <- NA # write a new metadata column for SMT depth
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1045"),18] <- 82 # assign SMT depths by core
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1048"),18] <- 320
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1068"),18] <- 108
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1069"),18] <- 138
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1070"),18] <- 69
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1081"),18] <- 56
sample_data(ps.frdp)$peakaom <- NA # creates a new metada column for peak AOM depth
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="PC1029"),19] <- 45.3 # assign peak AOM depths by core
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1045"),19] <- 65
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1048"),19] <- 313.5
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1068"),19] <- 108
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1069"),19] <- 137
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1070"),19] <- 67
sample_data(ps.frdp)[which(sample_data(ps.frdp)$core=="GC1081"),19] <- 56
sample_data(ps.frdp)$smtzposition <- "above" # creates default value "above" for samples 
sample_data(ps.frdp)[which(sample_data(ps.frdp)$geochem_zone=="below"),20] <- "below" # overwrites as "below" for samples where geochem_zone = below
sample_data(ps.frdp)$dist_above_peakaom <- sample_data(ps.frdp)$peakaom-sample_data(ps.frdp)$depth # creates a new variable, distance above peak AOM
```

### Pull out and subset taxonomic and relative abundance data
Find the most abundant taxonomic classes, pull out their relative abundances, create a dataframe of them. Then identify the most abundant ANME and SRB and do the same thing. Add metadata and finally subset by stage. 
```{r}
library(phyloseq)
library(ggplot2)

classes.ps.frdp <- unique(tax_table(ps.frdp)[,3]) # obtains the unique class names
classes.ps.frdp <- classes.ps.frdp[!is.na(classes.ps.frdp)] # removes the NA

mx.f.asv <- as.matrix(otu_table(ps.frdp)) # creates a matrix  from the phyloseq object
class.ra <- vector("list",0)
for (i in classes.ps.frdp) {class.ra[[i]] <- sum(mx.f.asv[,which(tax_table(ps.frdp)[,3]==i)])/31170.40}
class.output <- cbind(class.ra) 
# cab print list of classes by their relative abundance: print(class.output) 
top.c <- c("JS1","Deltaproteobacteria","Methanomicrobia","Dehalococcoidia","Anaerolineae","Campylobacteria","Aminicenantia","Bacteroidia","Phycisphaerae","Lokiarchaeia","Woesearchaeia","Gammaproteobacteria","Thermoplasmata","Alphaproteobacteria", "Nitrososphaeria") # all classes > 1% relabund
sum(25.57895,14.68791,10.22672,7.295351,4.331321,3.897544,3.558376,2.645234,2.289897,2.236513,1.741267,1.636488,1.2688,1.181987,1.01452) # 83.6% of sequences belong to these 15 classes. Sum of all 125 is probably 90.% (prior run flux_dada2)

f.ra <- transform_sample_counts(ps.frdp, function(OTU) OTU/sum(OTU)) # creates relative abundance phyloseq object
f.ra.top.c <- subset_taxa(f.ra, Class=="JS1" | Class=="Deltaproteobacteria" | Class=="Methanomicrobia" | Class=="Dehalococcoidia" | Class=="Anaerolineae" | Class=="Campylobacteria" | Class=="Aminicenantia" | Class=="Bacteroidia" | Class=="Phycisphaerae" | Class=="Lokiarchaeia" | Class=="Woesearchaeia" | Class=="Gammaproteobacteria" | Class=="Thermoplasmata" | Class=="Alphaproteobacteria" | Class=="Nitrososphaeria") # top 15 classes

# use a nested for-loop to pull relative abundance data out of phyloseq objects (by taxonomic division):
graphclass <- unique(tax_table(f.ra.top.c)[,3])
graphclass <- as.vector(graphclass)
class.sums.m <- matrix(nrow=76, ncol=15) # define matrix dimensions like this
rownames(class.sums.m) <- sample_names(f.ra.top.c) # specify row names
colnames(class.sums.m) <- graphclass # specify column names

for (i in sample_names(f.ra.top.c)) { 
  for (j in graphclass) {
        class.sums.m[[i,j]] <- sum(otu_table(f.ra.top.c)[sample_names(f.ra.top.c)==i,which(tax_table(f.ra.top.c)[,3]==j)]) # this is calculating the average percent abundance of each class
    }
}

# make data frames (in long format) from the matrix of abundances
flux.class.wide <- t(class.sums.m)
taxonrep <- rep(rownames(flux.class.wide), each=76)
tax <- as.character(taxonrep)
grouprep <- colnames(flux.class.wide)
groups <- rep(grouprep,times=15)
abund <- as.numeric(t(flux.class.wide))
longdf <- cbind.data.frame(tax,abund,groups)

# ANME and SRB info
# manually inspected taxonomy at different levels to decide which ANME families and SRB genera to include on the graph using variations of the commands directly below:
methanoASVs <- which(tax_table(f.ra)[,3]=="Methanomicrobia") # lists the ASVs that belong to Class Methanomicrobia
unique(tax_table(f.ra)[anmeASVs,5]) # shows family-level taxonomy of methano ASVs, see the top one is an ANME-1a
sum(otu_table(f.ra)[,which(tax_table(f.ra)[,5]=="ANME-1a")])/.76 # ANME-1a are 4.55% of sequences

# here's who I chose
ANMEs <- c("ANME-1a","ANME-1b","ANME-2c","ANME-2a-2b") # selected ANMEs at the Family level (Genus level contains many NAs)
SRBs <- c("SEEP-SRB1","SEEP-SRB2","Desulfatiglans","Desulfococcus","Desulfobulbus") # selected SRBs at the Genus level

flux.srbs <- subset_taxa(f.ra, Genus=="SEEP-SRB1" | Genus=="SEEP-SRB2" | Genus=="Desulfatiglans"| Genus=="Desulfococcus"| Genus=="Desulfobulbus" ) # contains 271 ASVs
flux.anme <- subset_taxa(f.ra, Family=="ANME-1a" | Family=="ANME-1b" | Family=="ANME-2a-2b" | Family=="ANME-2c") # contains 36 ASVs

# Pull out abundance data for ANMEs 
graphanme <- unique(tax_table(flux.anme)[,5])
graphanme <- as.vector(graphanme)
family.sums.m <- matrix(nrow=76, ncol=4) # define matrix dimensions 
rownames(family.sums.m) <- sample_names(flux.anme) # specify row names
colnames(family.sums.m) <- graphanme # specify column names
dim(family.sums.m) # matrix is 76 x 4

for (i in sample_names(flux.anme)) { 
  for (j in graphanme) {
        family.sums.m[[i,j]] <- sum(otu_table(flux.anme)[sample_names(flux.anme)==i,which(tax_table(flux.anme)[,5]==j)]) # this is calculating the average percent abundance of each ANME family
    }
}
head(family.sums.m)

# Pull out abundance data for SRBs
graphsrbs <- unique(tax_table(flux.srbs)[,6])
graphsrbs <- as.vector(graphsrbs)
gen.sums.m <- matrix(nrow=76, ncol=5) # define matrix dimensions 
rownames(gen.sums.m) <- sample_names(flux.srbs) # specify row names
colnames(gen.sums.m) <- graphsrbs # specify column names
dim(gen.sums.m) # matrix is 76 x 5 

for (i in sample_names(flux.srbs)) { 
  for (j in graphsrbs) {
        gen.sums.m[[i,j]] <- sum(otu_table(flux.srbs)[sample_names(flux.srbs)==i,which(tax_table(flux.srbs)[,6]==j)]) # this is calculating the average percent abundance of each SRB genus
    }
}
head(gen.sums.m)

# now turn these matrices into long dataframe form, combine them, and add metadata
# first for ANME
anme.taxonrep <- rep(colnames(family.sums.m), each=76)
anme.tax <- as.character(anme.taxonrep)
anme.grouprep <- rownames(family.sums.m)
anme.groups <- rep(anme.grouprep,times=4)
anme.abund <- as.numeric(family.sums.m)
longdf.family <- cbind.data.frame(anme.tax,anme.abund,anme.groups)
longdf.family$level <- "ANME Family"
longdf.family$Domain <- "Archaea"
longdf.family$tax_reorder <- factor(longdf.family$anme.tax, levels=c("ANME-1a","ANME-1b","ANME-2a-2b","ANME-2c"))

# now SRB
srb.taxonrep <- rep(colnames(gen.sums.m), each=76)
srb.tax <- as.character(srb.taxonrep)
srb.grouprep <- rownames(gen.sums.m)
srb.groups <- rep(srb.grouprep,times=5)
srb.abund <- as.numeric(gen.sums.m)
longdf.genus <- cbind.data.frame(srb.tax,srb.abund,srb.groups)
longdf.genus$level <- "SRB Genus"
longdf.genus$Domain <- "Bacteria"
longdf.genus$tax_reorder <- factor(longdf.genus$srb.tax, levels = c("SEEP-SRB1","SEEP-SRB2","Desulfatiglans","Desulfococcus","Desulfobulbus"))

# combine all the dataframes
# first longdf, 1140 rows
longdf$level <- "Class"
longdf$Domain <- "Bacteria" # writes default domain as Bacteria
archaearows <- which(longdf$tax=="Methanomicrobia"|longdf$tax=="Woesearchaeia"|longdf$tax=="Lokiarchaeia"|longdf$tax=="Nitrososphaeria") # specifies the rows where classes belong to Archaea.
longdf[archaearows,5] <- "Archaea" # overwrites domain from Bacteria to Archaea
longdf$tax_reorder <- factor(superdf$tax, levels = c("JS1","Dehalococcoidia","Anaerolineae","Aminicenantia","Campylobacteria","Bacteroidia","Phycisphaerae","Lokiarchaeia","Woesearchaeia","Gammaproteobacteria","Thermoplasmata","Alphaproteobacteria","Nitrososphaeria","Methanomicrobia","Deltaproteobacteria"))

colnames(longdf.genus)[1:3] <- c("tax", "abund", "groups") # rewrites column names so that they are not specific to SRB
colnames(longdf.family)[1:3] <- c("tax", "abund", "groups") # rewrites column names so that they are not specific to ANME
superdf2 <- rbind(longdf,longdf.family,longdf.genus)
superdf2$pctabund <- 100*superdf2$abund

# add in metadata
repmeta <- do.call("rbind", replicate(24, sample_data(f.ra.top.c), simplify = FALSE)) # repeats metadata for each sample to match the rows in the long dataframe. Use the number 24 because there are 24 total taxonomies to plot amongst three taxonomic groups (class, family, genus)
repmeta$groups <- row.names(repmeta)
metasuperdf <- merge.data.frame(repmeta,superdf2) # merges long data frame with abundances with the metadata
metasuperdf$level_reorder <- factor(metasuperdf$level, levels=c("Class","ANME Family","SRB Genus"))

# subset by core flowtype 
sdf1029 <- subset(metasuperdf, metasuperdf$stage=="seep") 
sdf.fi <- subset(metasuperdf, metasuperdf$stage=="fluxincreasing") 
sdf.ss <- subset(metasuperdf, metasuperdf$stage=="steadystate")
```

### Graph community composition bubble plots
Note that the Classes were all mixed up when I was plotting tax_reorder under aes(). Replotting by tax solves this, but I wanted to order Deltaproteobacteria and Methanomicrobia next to ANME and SRB. It's a small detail though.
```{r}
library(ggplot2)

# first the seep site
bp1029 <- ggplot(sdf1029,aes(tax,depth,size=pctabund))
bp.seep <- bp1029+
  geom_point(aes(fill=Domain),colour="black",pch=21)+
  xlab("")+
  scale_y_reverse("", limits=c(30,0), breaks=c(0,5,10,15,20,25,30))+
  scale_fill_discrete("")+
  facet_grid(~level_reorder, scales = "free", space = "free")+
  scale_size_area("% Abundance",breaks=c(1,5,10,20,40,60))+
  theme_bw()+
  theme(legend.position = "top", legend.box = "vertical", axis.text.x=element_text(angle = 55, hjust = 1,      
    size=11), axis.text.y=element_text(size = 12), legend.text=element_text(size=11),
    axis.title=element_text(size=14), strip.text = element_text(size=11), plot.margin = margin(0,0,0,-0.5,"cm"))
bp.seep

# methane flux increasing
bp.fi <- ggplot(sdf.fi,aes(tax,depth,size=pctabund))
bp.fluxincreasing <- bp.fi+
  geom_point(aes(fill=Domain),colour="black",pch=21)+
  xlab("")+
  facet_grid(core~level_reorder,scales="free", space = "free")+
  scale_y_reverse("",breaks=c(0,25,50,75,100,125,150),limits=c(150,0))+
  scale_fill_discrete("")+
  scale_size_area("% Abundance",breaks=c(1,5,10,20,40,60))+
  geom_hline(aes(yintercept = 50),linetype="dashed",size=0.8)+ # this line returns Error in -x : invalid argument to unary operator
  theme_bw()+
  theme(legend.position = "top", axis.text.x=element_text(angle = 55, hjust = 1, size=11),axis.text.y=element_text(size=12),legend.text=element_text(size=11),axis.title=element_text(size=14),strip.text.x = element_text(size=11), strip.text.y = element_blank(), plot.margin = margin(0,0,0,-0.5,"cm"), legend.box = "vertical")
bp.fluxincreasing 

# steady-state
bp.ss <- ggplot(sdf.ss,aes(tax,depth,size=pctabund))
bp.steadystate <- bp.ss+
  geom_point(aes(fill=Domain),colour="black",pch=21)+
  xlab("")+
  facet_grid(core~level_reorder,scales="free", space = "free")+
  scale_y_reverse("",breaks=c(0,50,100,150,200,250,300,350),limits=c(350,0))+
  scale_fill_discrete("")+
  scale_size_area("% Abundance",breaks=c(1,5,10,20,40,60))+
  geom_hline(aes(yintercept = smt),linetype="dashed",size=0.8)+
  theme_bw()+
  theme(legend.position = "top",axis.text.x=element_text(angle = 55, hjust = 1, size=11),axis.text.y=element_text(size=12),legend.text=element_text(size=11),axis.title=element_text(size=14),strip.text.x = element_text(size=11), strip.text.y = element_blank(), legend.box = "horizontal", plot.margin = margin(0,0,0,-0.5,"cm"))
bp.steadystate
```

### Import and graph porewater data
```{r}
library(ggplot2)
# import the porewater data
porewater <- read.csv(file="porewater.csv")

# subset by core flowtype 
pore.1029 <- subset(porewater, porewater$stage=="seep") 
pore.fi <- subset(porewater, porewater$stage=="inc") 
pore.ss <- subset(porewater, porewater$stage=="ss")

# seep site
ggp.1029 <- ggplot(pore.1029,aes(depth,value,color=species))
gg.porewater.1029 <- ggp.1029+geom_line(size=1.5)+
  geom_point(size=3)+
  coord_flip()+
  ggtitle("Porewater Geochemistry")+
  scale_y_continuous("",limits = c(0,40),position = "right")+
  scale_x_reverse("Depth (cmbsf)",breaks=c(0,5,10,15,20,25,30),limits=c(30,0))+
  scale_color_manual("",values=c("orangered1","black","dodgerblue2","gold3"),guide=guide_legend())+
  theme_bw()+
  theme(legend.position = "bottom", legend.direction = "vertical",axis.text=element_text(size=12),legend.text=element_text(size=11),axis.title=element_text(size=14))
gg.porewater.1029

# non-steady-state
ggp.fi <- ggplot(pore.fi,aes(depth,value,color=species))
gg.porewater.nss <- ggp.fi+geom_line(size=1.5)+
  geom_point(size=3)+
  geom_vline(aes(xintercept=smt),linetype="dashed",size=0.8)+
  coord_flip()+
  facet_grid(core~.)+
  scale_y_continuous("",limits = c(0,40),position = "right")+
  scale_x_reverse("Depth (cmbsf)",breaks=c(0,25,50,75,100,125,150),limits=c(150,0))+
  scale_color_manual("",values=c("orangered1","black","dodgerblue2","gold3"))+
  labs(title="Porewater Geochemistry")+
  theme_bw()+
  theme(axis.text=element_text(size=12),strip.text.y = element_text(size=12),title=element_text(size=11),axis.title=element_text(size=14),legend.text = element_text(size=12),legend.title = element_text(size=12),legend.direction = "vertical",legend.position = "bottom",plot.margin=margin(0,0,0,0.25,"cm"))
gg.porewater.nss

# steady-state
ggp.ss <- ggplot(pore.ss,aes(depth,value,color=species))
gg.porewater.ss <- ggp.ss+geom_line(size=1.5)+
  geom_point(size=3)+
  geom_vline(aes(xintercept=smt),linetype="dashed",size=0.8)+
  coord_flip()+
  facet_grid(core~.)+
  scale_y_continuous("",limits = c(0,40),position = "right")+
  scale_x_reverse("Depth (cmbsf)",breaks=c(0,50,100,150,200,250,300,350),limits=c(350,0))+
  scale_color_manual("",values=c("orangered1","black","dodgerblue2","gold3"))+
  labs(title="Porewater Geochemistry")+
  theme_bw()+
  theme(axis.text=element_text(size=12),strip.text.y = element_text(size=12),title=element_text(size=11),axis.title=element_text(size=14),legend.text = element_text(size=12),legend.title = element_text(size=12),legend.direction = "vertical",legend.position = "bottom",plot.margin=margin(0,0,0,0.25,"cm"))
gg.porewater.ss
```

### Import and graph modeled AOM rate data
Some of it I've found on this computer, but WeiLi's modeling of past years of 1045 and 1081 might be on my home computer
```{r}
aomrates <- read.csv(file="~/Desktop/svalflux/aomrates.csv") # import rate data
head(aomrates)
aomrates <- aomrates[,1:5] # trim off weird columns that contain no information

# subset by methane flux condition
aomrates.inc <- subset(aomrates, aomrates$core=="GC1045"|aomrates$core=="GC1081")
aomrates.ss <- subset(aomrates, aomrates$core=="GC1068"|aomrates$core=="GC1069"|aomrates$core=="GC1070"|aomrates$core=="GC1048")
aomrates.seep <- subset(aomrates, aomrates$core=="PC1029")

# graph seep rate
aom1029 <- ggplot(aomrates.seep, aes(depth,aom))
raom1029 <- aom1029+geom_line(size=1.5)+
  coord_flip()+
  scale_y_continuous("",position="right", breaks=c(0,2000,4000,6000,8000))+
  scale_x_reverse("",breaks=c(0,5,10,15,20,25,30),limits=c(30,0))+
  labs(title=expression("AOM rate (µmols L"^{-1}*" day"^{-1}*")"))+
  theme_bw()+
  theme(axis.text=element_text(size=12),legend.text=element_text(size=11),axis.title=element_text(size=14))
raom1029 

# graph ss rates
aom.ss <- ggplot(aomrates.ss, aes(depth,aom))
raom.ss <- aom.ss+geom_line(size=1)+
  geom_vline(aes(xintercept=smt),linetype="dashed",size=0.8)+
  coord_flip()+
  facet_grid(core~.)+
  scale_y_continuous("",limits = c(-5,600),position = "right")+
  scale_x_reverse("",breaks=c(0,50,100,150,200,250,300,350),limits=c(350,0))+
  theme_bw()+
  labs(title=expression("AOM rate \n(µmols L"^{-1}*" day"^{-1}*")"))+
  theme(axis.text=element_text(size=12),strip.text.y = element_blank(),strip.background = element_blank(),axis.text.y=element_blank(),title=element_text(size=11),plot.margin=margin(1,0,0,-0.25,"cm"))
raom.ss 

# graph inc rates 
aomrates.inc$yrsbp <- factor(aomrates.inc$yrsbp,levels=c(-2,-1,0,1,2,3,4,5,6,7,8,9,10)) 
head(aomrates.inc)

aom.nss <- ggplot(aomrates.inc,aes(depth,aom,color=yrsbp))
raom.nss <- aom.nss+geom_line(size=1)+
  scale_color_manual("years \nbefore \npresent",values=c("dodgerblue3","dodgerblue1","black","#662506","#993404","#cc4c02","#ec7014","#fe9929","#fec44f","grey75","grey65","grey55","grey45"))+
  geom_vline(aes(xintercept=smt),linetype="dashed",size=0.8)+
  coord_flip()+
  facet_grid(core~.)+
  labs(title=expression("AOM rate\n(µmols L"^{-1}*" day"^{-1}*")"))+
  scale_y_continuous("",breaks=c(0,100,200),limits = c(-5,220),position = "right")+
  scale_x_reverse("",breaks=c(0,25,50,75,100,125,150),limits=c(150,0))+
  theme_bw()+
  theme(axis.text=element_text(size=12),strip.text.y = element_blank(),strip.background = element_blank(),axis.text.y=element_blank(),title=element_text(size=11),plot.margin=margin(1,0,0,-0.25,"cm"))
raom.nss
```

### Graph ddPCR counts of mcrA and dsrAB
```{r}
# first make a dataframe with gene counts in long format
core <- rep(sample_data(ps.frdp)$core, times=2) # repeat core data for each gene
depth <- rep(sample_data(ps.frdp)$depth, times=2) # repeat depth 
smt <- rep(sample_data(ps.frdp)$smt, times=2) # repeat smt
genecount <- c(sample_data(ps.frdp)$dsrab, sample_data(ps.frdp)$mcra) # combine gene counts into one vector, mcra first
gene <- c(rep("dsrAB", each=76), rep("mcrA", each=76)) # create vector with type of gene being counted
gene <- factor(gene, levels = c("mcrA", "dsrAB"))
stage <- rep(sample_data(ps.frdp)$stage, times=2) # repeat stage
ddpcr.all <- data.frame(core, depth, smt, stage, genecount, gene) # make data frame
ddpcr.all$logcopies <- log10(ddpcr.all$genecount) # transform gene counts to log

# subset by stage of methane flux
dd.pcr.fi <- subset(ddpcr.all, ddpcr.all$stage=="fluxincreasing")
dd.pcr.ss <- subset(ddpcr.all, ddpcr.all$stage=="steadystate")
dd.pcr.seep <- subset(ddpcr.all, ddpcr.all$stage=="seep")

# seep ddPCR plot
dd1029 <- ggplot(dd.pcr.seep, aes(depth, logcopies, color=gene))
dd.seep <- dd1029+geom_line(size=1.5)+
  geom_point(size=3)+
  coord_flip()+
  scale_y_continuous("",breaks=c(4,5,6,7,8),limits = c(3.7,8.5),position="right")+
  scale_x_reverse("",breaks=c(0,5,10,15,20,25,30),limits=c(30,0))+
  scale_color_discrete("")+
  theme_bw()+
  labs(title=expression(paste("log"[10]," gene copies")))+
  theme(legend.position = "top",axis.text=element_text(size=12),legend.text=element_text(size=11),axis.title=element_text(size=8))
dd.seep
# remove the NAs at 12 cm depth that prevent the lines from being connected

# non-steady-state ddPCR plot
ddfi<- ggplot(dd.pcr.fi, aes(depth, logcopies, color=gene))
dd.fi <- ddfi+geom_line(size=1.5)+
  geom_point(size=3)+
  coord_flip()+
  scale_y_continuous("", breaks=c(4,5,6,7), limits = c(3.7,7.4), position="right")+
  scale_x_reverse("", breaks=c(0,25,50,75,100,125,150), limits=c(150,0))+
  scale_color_discrete("")+
  geom_vline(aes(xintercept=smt),linetype="dashed",size=0.8)+
  facet_grid(core~.)+
  theme_bw()+
  labs(title=expression(paste("log"[10]," gene copies")))+
  theme(legend.position = "top",axis.text=element_text(size=12),legend.text=element_text(size=11),axis.title=element_text(size=8))
dd.fi

# steady-state ddPCR plot
ddss<- ggplot(dd.pcr.ss, aes(depth, logcopies, color=gene))
dd.ss <- ddss+geom_line(size=1.5)+
  geom_point(size=3)+
  coord_flip()+
  scale_y_continuous("", breaks=c(4,5,6,7), limits = c(3.7,7.4), position="right")+
  scale_x_reverse("", breaks=c(0,50,100,150,200,250,300,350), limits=c(350,0))+
  geom_vline(aes(xintercept=smt), linetype="dashed",size=0.8)+
  facet_grid(core~.)+
  theme_bw()+
  labs(title=expression(paste("log"[10]," gene copies")))+
  theme(legend.position = "top",axis.text=element_text(size=12),legend.text=element_text(size=11),axis.title=element_text(size=8))
dd.ss
# need to remove the samples with zeros that obscure the trend or at least make the graph look really chaotic
```

### Make panel figures
```{r}
library(egg)

# Figure 2 (Seep)
fig1029 <- ggarrange(gg.porewater.1029, raom1029, bp.seep, dd.seep, widths=c(1.1,0.9,2.5,0.7),ncol = 4,nrow = 1,labels=c("A","B","C","D"))
fig1029

# Figure 3 (Non-steady-state sites)
fig.nss <- ggarrange(gg.porewater.nss, raom.nss, bp.fluxincreasing, dd.fi, ncol = 4, nrow = 1,
  widths = c(2.4,2.1,5,1.8),labels=c("A","B","C","D"))
fig.nss

# Figure 4 (Steady-state sites)
fig.ss <- ggarrange(gg.porewater.ss, raom.ss, bp.steadystate, dd.ss,
                        ncol = 4, nrow = 1,widths = c(2.4,1.8,4.8,2),labels=c("A","B","C","D"))
fig.ss
```

### Ordinations and microbial community comparisons
```{r}

```

### DEseq
```{r}

```

### Supplemental figures
Total methane flux across all cores
Regression of mcrA counts and modeled AOM rates
Justification of steady-state vs non-steady-state
I think I'm forgetting one...
```{r}

```


